### PR TITLE
Updated dependencies for SDK generation feature of APIM Store

### PR DESCRIPTION
**Pull request by vlgunarathne**
This PR is for the dependency updates that are required for the SDK generation feature for APIs in the API Manager Store.

The following are the upgraded dependencies.

- snakeyaml/1.18.0.wso2v1
- swagger-codegen/2.2.3.wso2v1
- swagger-parser/1.0.32.wso2v1

### CODE REVIEW

File: snakeyaml/1.18.0.wso2v1/pom.xml [hunk 1]
```diff
@@ -0,0 +1,83 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+ ~ Copyright (c) 2015, WSO2 Inc. (http://wso2.com) All Rights Reserved.
```
Comment (cnapagoda) [inline snakeyaml/1.18.0.wso2v1/pom.xml]: License year is wrong

File: swagger-parser/1.0.32.wso2v1/pom.xml [hunk 2]
```diff
@@ -0,0 +1,112 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+ ~ Copyright (c) 2016, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (cnapagoda) [inline swagger-parser/1.0.32.wso2v1/pom.xml]: License year is wrong

File: snakeyaml/1.18.0.wso2v1/pom.xml [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+    </dependencies>
+
+    <distributionManagement>
+        <repository>
+            <id>wso2.releases</id>
+            <name>WSO2 internal Repository</name>
+            <url>http://maven.wso2.org/nexus/content/repositories/releases/</url>
+        </repository>
+
+        <snapshotRepository>
```
Comment (thusithathilina) [inline snakeyaml/1.18.0.wso2v1/pom.xml]: remove snapshot distro

File: swagger-codegen/2.2.3.wso2v1/pom.xml [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+        <dependency>
+            <groupId>io.swagger</groupId>
+            <artifactId>swagger-codegen</artifactId>
+            <version>2.2.3</version>
+            <optional>true</optional>
+        </dependency>
+        <dependency>
+            <groupId>com.samskivert</groupId>
+            <artifactId>jmustache</artifactId>
+            <version>1.13</version>
```
Comment (thusithathilina) [inline swagger-codegen/2.2.3.wso2v1/pom.xml]: All the dependencies should be optional

File: swagger-codegen/2.2.3.wso2v1/pom.xml [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+        </dependency>
+        <dependency>
+            <groupId>joda-time</groupId>
+            <artifactId>joda-time</artifactId>
+            <version>2.9.9</version>
+        </dependency>
+        <dependency>
```
Comment (thusithathilina) [inline swagger-codegen/2.2.3.wso2v1/pom.xml]: Need formatting

File: swagger-parser/1.0.32.wso2v1/pom.xml [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+                            io.swagger.models; version="[1.5.8,1.6)",
+                            com.fasterxml.jackson.databind.type; version="[2.7.0,3.0.0)",
+                            com.fasterxml.jackson.databind.node; version="[2.7.0,3.0.0)",
+                            com.fasterxml.jackson.databind; version="[2.7.0,3.0.0)",
+                            org.apache.commons.io;version="1.4",
+                            org.apache.commons.io.comparator;version="1.4",
+                            org.apache.commons.io.filefilter;version="1.4",
+                            org.apache.commons.io.input;version="1.4",
+                            org.apache.commons.io.output;version="1.4",
+                            org.yaml.snakeyaml.*;version="1.16.0.wso2v1"
```
Comment (thusithathilina) [inline swagger-parser/1.0.32.wso2v1/pom.xml]: Why depend on wso2 version?

File: swagger-codegen/2.2.3.wso2v1/pom.xml [hunk 7]
```diff
@@ -0,0 +1,167 @@
...
+            <optional>true</optional>
+        </dependency>
+        <dependency>
+            <groupId>joda-time</groupId>
+            <artifactId>joda-time</artifactId>
+            <version>2.9.9</version>
+            <optional>true</optional>
+        </dependency>
+        <dependency>
+		    <groupId>org.codehaus.plexus</groupId>
```
Comment (thusithathilina) [inline swagger-codegen/2.2.3.wso2v1/pom.xml]: Need formatting

**Pull request closed without merging**